### ISSUE TITLE
SIGABRT on BUFS_Build()

### ISSUE THREAD

**Issue by mdukat** [d3suu/BUFS#2]
Basically for some unknown reason BUFS_Build() throws SIGABRT.

strace:
```
write(1, "Writed 57 files to test.img.\n", 29Writed 57 files to test.img.
) = 29
writev(2, [{iov_base="corrupted size vs. prev_size", iov_len=28}, {iov_base="\n", iov_len=1}], 2corrupted size vs. prev_size
) = 29
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f5f1fb31000
rt_sigprocmask(SIG_UNBLOCK, [ABRT], NULL, 8) = 0
rt_sigprocmask(SIG_BLOCK, ~[RTMIN RT_1], [], 8) = 0
getpid()                                = 9439
gettid()                                = 9439
tgkill(9439, 9439, SIGABRT)             = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
--- SIGABRT {si_signo=SIGABRT, si_code=SI_TKILL, si_pid=9439, si_uid=1000} ---
+++ killed by SIGABRT (core dumped) +++
Aborted (core dumped)
```

gdb backtrace:
```
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
#1  0x00007ffff747e801 in __GI_abort () at abort.c:79
#2  0x00007ffff74c7897 in __libc_message (action=action@entry=do_abort, fmt=fmt@entry=0x7ffff75f4b9a "%s\n") at ../sysdeps/posix/libc_fatal.c:181
#3  0x00007ffff74ce90a in malloc_printerr (str=str@entry=0x7ffff75f2c9d "corrupted size vs. prev_size") at malloc.c:5350
#4  0x00007ffff74d615f in _int_free (have_lock=0, p=<optimized out>, av=0x7ffff7829c40 <main_arena>) at malloc.c:4295
#5  __GI___libc_free (mem=<optimized out>) at malloc.c:3124
#6  0x00007ffff751dbbd in __closedir (dirp=<optimized out>) at ../sysdeps/posix/closedir.c:50
#7  0x00005555555556c0 in BUFS_Build(char*, char*) ()
#8  0x0000555555555a07 in main ()
```

As far as i can see, the abort() occurs in __closedir(), which can mean that BUFS_Build() have some problems with closing the output (image) file **OR** there is problem in closedir() function, becouse SIGABRT occurs after `printf("Writed %d files to %s.\n", filesNumber, imagePath);`.

However this bug isn't super-duper serious, becouse output file is written correctly, and no directory is locked, so its *probably* safe.

**Comment by mdukat** [comment]
Based on [closedir.c](https://code.woboq.org/userspace/glibc/sysdeps/posix/closedir.c.html):50, just the `free(void*)` have some problem with `DIR* dir` pointer, however its not used anymore in this case, so it will be ignored.

TLDR its time to build signal handler which will be ignoring SIGABRT in case of `BUFS_Build()` function.

**Comment by mdukat** [comment]
Partly fixed in [commit 55d00efecba911e248c6e07eb2e35af8ee22fc1a](https://github.com/d3suu/BUFS/commit/55d00efecba911e248c6e07eb2e35af8ee22fc1a).

Closing due to possible internal bug in glibc.
